source/fundamentals/data-formats/polymorphism.txt

```diff
@@ -0,0 +1,271 @@
+   BsonClassMap.RegisterClassMap<Tiger>();
+
+Use Discriminators
+------------------
+
+In MongoDB, a **discriminator** is a field added to a document to identify the class
+to which the document deserializes. When a collection contains more than one type from a
+single inheritance hierarchy, discriminators ensure that each
+document deserializes to the right class. The {+driver-short+} stores the discriminator
+value in a field named ``_t`` in the BSON document. Generally, ``_t`` is the second
+field in the BSON document after ``_id``.
+
+**Discriminator conventions** define the value stored in the discriminator field.
+In this section, you can learn about the discriminator conventions included
+with the {+driver-short+} and how to create custom discriminator conventions.
+
+ScalarDiscriminatorConvention
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the {+driver-short+} uses the ``ScalarDiscriminatorConvention``. According
+to this convention, the {+driver-short+} sets the value of the ``_t`` field to the name of
+the class from which the document was serialized.
+
+Suppose you created an instance of the example ``Animal`` class and each of its
+subclasses. If you serialized these objects to a single collection, the {+driver-short+}
+would apply the ``ScalarDiscriminatorConvention`` and the corresponding
+BSON documents would appear as follows:
+
+.. code-block:: json
+   :copyable: false
+   
+   { _id: ..., _t: "Animal", ... }
+   { _id: ..., _t: "Cat", ... }
+   { _id: ..., _t: "Dog", ... }
+   { _id: ..., _t: "Lion", ... }
+   { _id: ..., _t: "Tiger", ... }
+
+The ``ScalarDiscriminatorConvention`` uses concise discriminator values, but can be
+difficult to run a query on. For example, to find all documents of type or subtype ``Cat``,
+you must explicitly list each class you're looking for:
+
+.. code-block:: csharp
+   :copyable: true
+   
+   var query = coll.Aggregate().Match(a => a is Cat || a is Lion || a is Tiger);
+
+HierarchicalDiscriminatorConvention
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To simplify queries against your collection of polymorphic types, you can use the
+``HierarchicalDiscriminatorConvention``. According to this convention, the value of ``_t``
+is an array of all classes in the inheritance hierarchy of the document's type.
+
+To use the ``HierarchicalDiscriminatorConvention``, label the base class of your
+inheritance hierarchy as the root class. If you're using the automapper,
+label the root class by applying the
+``[BsonDiscriminatorAttribute]`` attribute to the class and passing ``RootClass = true``
+as an argument. The following code example labels the ``Animal`` class as the
+root of the example inheritance hierarchy:
+
+.. code-block:: csharp
+   :emphasize-lines: 1
+   
+   [BsonDiscriminator(RootClass = true)]
+   [BsonKnownTypes(typeof(Cat), typeof(Dog)]
+   public class Animal 
+   {
+   }
+
+If you're creating a class map manually, call the ``SetIsRootClass()`` method and pass
```

Comment:
You've used this consistently, I just want to confirm that the method is `SetIsRootClass()`, rather than `SetAsRootClass()`.